files/default/configure_nw_interface.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh
+
+# expected vars:
+# DEVICE_NAME: name of the device
+# DEVICE_NUMBER: number of the device
+# GW_IP_ADDRESS: gateway ip address
+# DEVICE_IP_ADDRESS: ip address to assign to the interface
+# CIDR_PREFIX_LENGTH: the prefix length of the device IP cidr block
+# NETMASK: netmask to apply to device ip address
+
+ROUTE_TABLE="100${DEVICE_NUMBER}"
+
+# config file
```

Comment:
I'd add an explanation of the various steps.
* What is the scope of this ifcfg file?
* Why do we need to create a route file and why are we setting a "metric"?
* Why do we need a rule file?